### PR TITLE
Refactoring - Decoupling Type System from Compiler

### PR DESCRIPTION
**Pull request by tgiphil**

### CODE REVIEW

File: Source/Mosa.Utility.FileSystem/Mosa.Utility.FileSystem.csproj [hunk 1]
```diff
@@ -3,7 +3,8 @@
     <TargetFramework>net7.0</TargetFramework>
     <LangVersion>11.0</LangVersion>
     <OutputPath>..\..\bin\</OutputPath>
-    <AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
+	<Copyright>Copyright © MOSA Project 2023</Copyright>
```
Comment (charsleysa) [inline Source/Mosa.Utility.FileSystem/Mosa.Utility.FileSystem.csproj:6]: mixed spaces and tabs

**Review by charsleysa — APPROVED** [review]
LGTM

File: Source/Mosa.Utility.FileSystem/Mosa.Utility.FileSystem.csproj (hunk 1 above)
Comment (tgiphil) [reply Source/Mosa.Utility.FileSystem/Mosa.Utility.FileSystem.csproj:6]: Fixed in next update.

File: Source/Mosa.Compiler.MosaTypeSystem/BuiltInTypes.cs [hunk 2]
```diff
@@ -65,10 +65,10 @@ public BuiltInTypes(ITypeResolver typeResolver, MosaModule corlib)
 		Object = typeResolver.ResolveType(corlib, BuiltInType.Object);
 		I = typeResolver.ResolveType(corlib, BuiltInType.IntPtr);
 		U = typeResolver.ResolveType(corlib, BuiltInType.UIntPtr);
-		TypedRef = typeResolver.ResolveType(corlib, BuiltInType.TypedReference);
 
-		Pointer = Void.ToUnmanagedPointer();
-		ManagedPointer = Void.ToManagedPointer();
+		//TypedRef = typeResolver.ResolveType(corlib, BuiltInType.TypedReference);
+		//Pointer = Void.ToUnmanagedPointer();
+		//ManagedPointer = Void.ToManagedPointer();
```
[the remark this answers is not in the document]
Comment (tgiphil) [reply Source/Mosa.Compiler.MosaTypeSystem/BuiltInTypes.cs:71]: by design

File: Source/Mosa.Compiler.Framework/PrimitiveType.cs [hunk 3]
```diff
@@ -0,0 +1,6 @@
+﻿// Copyright (c) MOSA Project. Licensed under the New BSD License.
+
+namespace Mosa.Compiler.Framework;
+
+public enum PrimitiveType
+{ Int32, Int64, R4, R8, Object, ManagedPointer, ValueType };
```
[the remark this answers is not in the document]
Comment (tgiphil) [reply Source/Mosa.Compiler.Framework/PrimitiveType.cs:6]: Optional

File: Source/Mosa.Compiler.Framework/Transforms/Call/CallDynamic.cs [hunk 4]
```diff
@@ -23,14 +23,13 @@ public override void Transform(Context context, TransformContext transform)
...
 		var operands = context.GetOperands();
 
 		operands.RemoveAt(0);
 		context.Empty();
 
-		MakeCall(transform, context, call, result, operands, method);
+		MakeCall(transform, context, call, result, operands);
 
-		transform.MethodScanner.MethodInvoked(call.Method, method);
+		//transform.MethodScanner.MethodInvoked(call.Method, method);
```
[the remark this answers is not in the document]
Comment (tgiphil) [reply Source/Mosa.Compiler.Framework/Transforms/Call/CallDynamic.cs:33]: This is okay.

**Pull request merged**